packages/website/blog/2024-11-30-catching-any.mdx

```diff
@@ -0,0 +1,237 @@
+  // eslint(@typescript-eslint/no-unsafe-argument):
+  // Unsafe assignment of an `any` value.
+}
+```
+
+### Banning Unsafe Calls
+
+[`@typescript-eslint/no-unsafe-call`](https://typescript-eslint.io/rules/no-unsafe-call) reports on assigning an `any` typed value to a property or variable.
+
+For example, given some value returned as `any` by `JSON.parse`, the rule would report on it being assigned to an object's property:
+
+```ts
+const person = {
+  name: '(anonymous)',
+};
+
+function setNameFromData(raw: string) {
+  person.name = JSON.parse(raw);
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // eslint(@typescript-eslint/no-unsafe-argument):
+  // Unsafe assignment of an `any` value.
+}
+```
+
+### Banning Unsafe Member Accesses
+
+[`@typescript-eslint/no-unsafe-member-access`](https://typescript-eslint.io/rules/no-unsafe-member-access)
+
+### Banning Unsafe Returns
+
+[`@typescript-eslint/no-unsafe-return`](https://typescript-eslint.io/rules/no-unsafe-return)
+
+Each of those rules target a different unsafe practice around the `any` type
+
+## Additional Helpers
+
+### Disabling TypeScript Suppressions
+
+The `any` type is not the only way to bypass TypeScript's type system.
+Developers are also able to use inline TypeScript directives: `// @ts-expect-error` and `// @ts-ignore`.
+Those inline comment directives cause TypeScript to ignore type errors on a line.
+Disabling TypeScript on a per-line basis can sometimes be necessary in rare cases, but is always unsafe and almost never the right choice.
+
+:::tip
+`// @ts-expect-error` and `// @ts-ignore` are almost the same, except `// @ts-expect-error` will produce a new type error if its line doesn't already have one.
+It's generally preferable to use `// @ts-expect-error` over `// @ts-ignore`.
+:::
+
+[`@typescript-eslint/ban-comment`](/rules/ban-ts-comment) can be used to report on cases where developers use TypeScript comment directives.
+By default, the rule:
+
+- Always reports on `@ts-ignore` and `@ts-nocheck` comment directives
+- Reports on `@ts-expect-error` directives that don't have an explanatory comment description
+- Requires explanatory comment descriptions contain at least 3 characters
+
+For example, suppose `"@example/package"` exports a `process` function that should take in a `string` but whose types incorrectly indicate take in a `number`.
+A `// @ts-expect-error` comment could be used to tell TypeScript to ignore the line:
+
+<Tabs>
+<TabItem value="❌ Incorrect">
+
+```ts
+import { process } from '@example/package';
+
+// @ts-ignore
+process('New York City');
+```
+
+</TabItem>
+<TabItem value="✅ Correct">
+
+```ts
+import { process } from '@example/package';
+
+// @ts-expect-error -- Pending updating the process types. See GH-1234.
+process('New York City');
+```
+
+</TabItem>
+</Tabs>
+
+One the types for `@example/package` are fixed to no longer produce a type error, the comment directive will itself produce a type error asking to delete itself.
```

Comment:
I don't think this blog post should explain how `// @ts-expect-error` / `// @ts-ignore` work. I think that'd be better served in a separate post on a TypeScript-dedicated page.

I haven't found one online I truly like so I might just put it on learningtypescript.com... the ideal state would be to have a dedicated page on the first-party typescriptlang.org docs IMO.